webSites/webSites-war/src/main/java/org/silverpeas/components/websites/servlets/WebSitesRequestRouter.java

```diff
@@ -90,1056 +87,1137 @@ public WebSiteSessionController createComponentSessionController(
-          scc.createFolder(currentPath + "/" + name);
+    FolderDetail webSitesCurrentFolder = scc.getFolder(id);
+    scc.setSessionTopic(webSitesCurrentFolder);
+    request.setAttribute("CurrentFolder", webSitesCurrentFolder);
 
-          request.setAttribute("Site", scc.getSessionSite());
+    return "/webSites/jsp/organize.jsp?Action=" + action + "&Id=" + id + "&Path=" + path;
+  }
 
-        } else if ("renameFolder".equals(action)) {
-          // = null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
-          // = changement de noms des fichiers et repertoires
-          String newName = request.getParameter("newName");
+  private String modifySiteDescription(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
+    String id = request.getParameter("Id");
+    // = null ou rempli si type= design
+    String currentPath = request.getParameter("path");
+    String type = request.getParameter("type"); // null ou design
+
+    request.setAttribute("Site", scc.getWebSite(id));
+    request.setAttribute("AllIcons", scc.getAllIcons());
+    request.setAttribute("ListIcons", scc.getIcons(id));
+
+    String recupParam = request.getParameter("RecupParam"); // =null ou oui
+    String complete = null;
+    if (recupParam != null) {// =oui
+      String nom = request.getParameter("Nom");
+      String description = request.getParameter("Description");
+      String lapage = request.getParameter("Page");
+      String listeIcones = request.getParameter("ListeIcones");
+
+      type = "design";
+      complete =
+          "&RecupParam=oui&Nom=" + nom + "&Description=" + description + "&Page=" + lapage +
+              "&ListeIcones=" + listeIcones;
+    } else {
+      return
+          "/webSites/jsp/modifDesc.jsp?Id=" + id + "&path=" + currentPath + "&type=" + type;
+    }
 
-          // RENAME FOLDER -------------------------------------------------------------
+    String destination = "/webSites/jsp/modifDesc.jsp?Id=" + id;
+    if (complete != null) {
+      destination += complete;
+    }
+    if (currentPath != null) {
+      destination += "&path=" + currentPath;
+    }
+    if (type != null) {
+      destination += "&type=" + type;
+    }
+    return destination;
+  }
 
-          /* Modifier le nom du repertoire */
-          scc.renameFolder(currentPath + "/" + name, currentPath + "/" + newName);
+  private String moveDownPublication(final WebSiteSessionController scc, final HttpRequest request,
+      final String userRole) throws WebSitesException {
+    String pubId = request.getParameter("Id");
 
-          request.setAttribute("Site", scc.getSessionSite());
+    scc.changePubsOrder(pubId, 1);
 
-        } else if ("deleteFolder".equals(action)) {
-          // null la premiere fois, puis = nom du repertoire courant
-          String name = request.getParameter("name");
+    S
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Publisher" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Move the "Admin" string literal on the left side of this string comparison. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1132)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Admin" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Define a constant instead of duplicating this literal "Publisher" 3 times. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1192)